jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -83,3 +85,59 @@ class SpectrumPerSpaxel(SinglePixelRegion):
     tool_id = 'jdaviz:spectrumperspaxel'
     action_text = 'See spectrum at a single spaxel'
     tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._spectrum_viewer = None
+        self._previous_bounds = None
+        self._mark = None
+        self._data = None
+
+    def _reset_spectrum_viewer_bounds(self):
+        sv_state = self._spectrum_viewer.state
+        sv_state.x_min = self._previous_bounds[0]
+        sv_state.x_max = self._previous_bounds[1]
+        sv_state.y_min = self._previous_bounds[2]
+        sv_state.y_max = self._previous_bounds[3]
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_move, events=['mousemove', 'mouseleave'])
+        if self._spectrum_viewer is None:
+            self._spectrum_viewer = self.viewer.jdaviz_helper.app.get_viewer('spectrum-viewer')
+        if self._mark is None:
+            self._mark = PluginLine(self._spectrum_viewer, visible=False)
+            self._spectrum_viewer.figure.marks = self._spectrum_viewer.figure.marks + [self._mark,]
+        # Store these so we can revert to previous user-set zoom after preview view
+        sv_state = self._spectrum_viewer.state
+        self._previous_bounds = [sv_state.x_min, sv_state.x_max, sv_state.y_min, sv_state.y_max]
+        super().activate()
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_move)
+        self._reset_spectrum_viewer_bounds()
+        super().deactivate()
+
+    def on_mouse_move(self, data):
+        if data['event'] == 'mouseleave':
+            self._mark.visible = False
+            self._reset_spectrum_viewer_bounds()
+            return
+
+        x = int(np.round(data['domain']['x']))
+        y = int(np.round(data['domain']['y']))
+
+        # Use first visible layer for now
+        cube_data = [layer.layer for layer in self.viewer.layers if layer.state.visible][0]
```

Comment:
should this use similar logic as #2646 to ensure its pulling from the flux cube?  What happens when there are no visible layers (I'm guessing a traceback... we might want to just skip that case gracefully)?

Reply:
I changed this to return on an empty result, good catch.

Reply:
The logic in 2646 might be useful, but I don't think it's worth holding this up for that to be merged. We could potentially do that if/when we change the axis handling.

Reply:
ok, but the first visible layer now could still be a flat image, or the uncertainty or mask cubes, right?

Reply:
I added a check to at least only use 3D data, avoiding a traceback if something like a collapsed image is displayed.

Reply:
thinking about this more... I think this should listen to the layer selection in mouseover itself, which also might fix the bug with the uncertainty cube failing to display.  That or we just ALWAYS pull the flux cube, regardless of the viewer and whether it is an active layer in that viewer or not 🤷